src/operator/src/statement/copy_table_from.rs

```diff
@@ -379,11 +378,23 @@ impl StatementExecutor {
 
         let mut rows_inserted = 0;
         let mut insert_cost = 0;
-        let max_insert_rows = req
-            .with
-            .get(MAX_INSERT_ROWS)
-            .and_then(|val| val.parse::<usize>().ok())
-            .unwrap_or(DEFAULT_MAX_INSERT_ROWS);
+        let max_insert_rows = if let Some(num) = req.limit {
+            Some(num as usize)
+        } else {
+            match req.with.get(MAX_INSERT_ROWS) {
+                Some(num) => match num.parse::<usize>() {
+                    Ok(num) => Some(num),
+                    Err(_) => {
+                        return Err(error::InvalidCopyParameterSnafu {
+                            key: MAX_INSERT_ROWS,
+                            value: num,
+                        }
+                        .build());
+                    }
+                },
```

Comment:
We could further simplify the code
```suggestion
                Some(num) => Some(num.parse::<usize>()
                        .context(error::InvalidCopyParameterSnafu {
                            key: MAX_INSERT_ROWS,
                            value: num,
                        })?),
```

Reply:
You could use [ok()](https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.ok) to convert the result into an option.